packages/mermaid/src/diagrams/pie/pieDb.ts

```diff
@@ -0,0 +1,99 @@
+import { log } from '../../logger.js';
+import { parseDirective as _parseDirective } from '../../directiveUtils.js';
+import { getConfig as commonGetConfig } from '../../config.js';
+import { sanitizeText } from '../common/common.js';
+import {
+  setAccTitle,
+  getAccTitle,
+  setDiagramTitle,
+  getDiagramTitle,
+  getAccDescription,
+  setAccDescription,
+  clear as commonClear,
+} from '../../commonDb.js';
+import type { ParseDirectiveDefinition } from '../../diagram-api/types.js';
+import type { PieFields, PieDB, Sections } from './pieTypes.js';
+import type { RequiredDeep } from 'type-fest';
+import type { PieDiagramConfig } from '../../config.type.js';
+import { cleanAndMerge } from '../../utils.js';
+
+export const DEFAULT_PIE_CONFIG: Required<PieDiagramConfig> = {
+  useMaxWidth: true,
+  useWidth: 984,
+  textPosition: 0.75,
+} as const;
+
+export const DEFAULT_PIE_DB: RequiredDeep<PieFields> = {
+  sections: {},
+  showData: false,
+  config: DEFAULT_PIE_CONFIG,
+} as const;
+
+let sections: Sections = DEFAULT_PIE_DB.sections;
+let showData: boolean = DEFAULT_PIE_DB.showData;
+let config: Required<PieDiagramConfig> = structuredClone(DEFAULT_PIE_CONFIG);
+
+const setConfig = (conf: PieDiagramConfig): void => {
+  config = cleanAndMerge(DEFAULT_PIE_CONFIG, conf);
+};
+
+const getConfig = (): Required<PieDiagramConfig> => config;
+
+const resetConfig = (): void => {
+  config = structuredClone(DEFAULT_PIE_CONFIG);
+};
```

Comment:
```suggestion
let config: Required<PieDiagramConfig> = structuredClone(DEFAULT_PIE_CONFIG);

const setConfig = (conf: PieDiagramConfig): void => {
  config = cleanAndMerge(DEFAULT_PIE_CONFIG, conf);
};

const getConfig = (): Required<PieDiagramConfig> => config;

const resetConfig = (): void => {
  config = structuredClone(DEFAULT_PIE_CONFIG);
};
```

I'm pretty against the PieDb having it's own `setConfig()`/`resetConfig()` functions.

I feel like this is just going to bugs like https://github.com/mermaid-js/mermaid/issues/4345 harder to fix.

Does this need to be in this PR? I feel like if we do want to make this change, it should at least be in it's own PR so we can discuss it separately from all the other changes.

Reply:
Removed for now, here is the changes so when we need to re-add it: https://github.com/mermaid-js/mermaid/compare/cb5f70c1396671d1570651b99b9ce79f27c60a2c...3f3a7340e3e282443290af61198f25fb3e6e1964